shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1109,9 +1125,23 @@ public void onStop() {
     }
   }
 
+  @Override
+  @TargetApi(18)
+  @RequiresApi(18)
+  public void onViewCreated(View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    view.getViewTreeObserver().addOnWindowFocusChangeListener(onWindowFocusChangeListener);
```

Comment:
You also need to guard this by an API level check. Adding `@RequiresApi(18)` to an Android lifecycle method is a bad idea. But I don't think this is the correct solution. You said Flutter only supports Android 19+, so if you're running into API level issues, you likely just need to change some `minSdkVersion` values wherever you're hitting the issue. Probably this [AndroidManifest](https://github.com/flutter/engine/blob/main/shell/platform/android/AndroidManifest.xml) and possibly also in some test code.

Reply:
Yes, it seems that while we do only officially support only 19+, the migration of our build environment is still in progress, so we need the min version checks until all the `minSdkVersion`s have been bumped.

See https://github.com/flutter/flutter/pull/125515

Reply:
In that case, you should not put `@RequiresApi` on `onViewCreated`, and instead add an explicit API level check as you did on `onDestroyView`. You may also want to manually verify that the field declaration using `@RequiresApi` doesn't break anything on Android 17 or below. My instinct is that it shouldn't as long as the field is otherwise unreferenced, but I'm not 100% positive.

Reply:
Just confirmed with an Android framework expert that the field will definitely be problematic. You should be able to solve that by initializing the value like 
```
Build.VERSION.SDK_INT >= 18 ? new OnWindowFocusChangeListener() { ... } : null
```

Reply:
Should I continue to use `@TargetApi(18)` for `onViewCreated`, now that I've got the version check? Or is it redundant now that the `onViewCreated` doesn't have `@RequiresApi`? There's no warning to suppress.

Reply:
The advice I've heard from AndroidX/Android Framework teams is never use `@TargetApi` at all. But you shouldn't apply either to `onViewCreated` since you've applied a runtime internal version check.